src/openrct2-ui/windows/EditorObjectSelection.cpp

```diff
@@ -793,7 +793,7 @@ class EditorObjectSelectionWindow final : public Window
 
     void OnTextInput(WidgetIndex widgetIndex, std::string_view text) override
     {
-        if (widgetIndex != WIDX_FILTER_TEXT_BOX || text.empty())
+        if (widgetIndex != WIDX_FILTER_TEXT_BOX)
```

Comment:
What's the reason for this change?

Reply:
You can test this yourself in an official build. In the object selection window, just enter a single character into the filter text box. You will notice the handle on the scroll bar expands, indicating that items have been filtered from the list. If you then hit backspace, the text box will now be empty, but the scroll bar handle will not shrink back to it's original size. The old filter string is still being used and will continue to be used until either a different string is entered or the clear button is pressed.

Reply:
Same bug in `TrackList.cpp`

Reply:
This belongs in a PR of its own.

Reply:
I've removed this change and opened #19112 to track this bug